UnicodeJsps/src/main/java/org/unicode/jsp/UnicodeJsp.java

```diff
@@ -208,28 +208,39 @@ public static void getDifferences(
         UnicodeUtilities.getDifferences(setA, setB, abbreviate, abResults, abSizes, abLinks);
     }
 
-    public static int parseCode(String text, String nextButton, String previousButton) {
+    public static int[] parseCode(String text, String nextButton, String previousButton) {
         // text = fromHTML.transliterate(text);
         String trimmed = text.trim();
-        if (trimmed.length() > 1) {
+        // Accept U+ notation and standalone hexadecimal digits, as well as a variety
+        // of hexadecimal character escape and numeric literal syntaxes.
+        Matcher matcher =
+                Pattern.compile(
+                                "(?:U\\+|\\\\[ux]\\{?|&#x|0x|16#|&H)?([0-9a-f'_]+)[\\};#]?",
+                                Pattern.CASE_INSENSITIVE)
+                        .matcher(trimmed);
+        String digits = matcher.matches() ? matcher.group(1).replaceAll("['_]", "") : null;
+        if (digits != null && digits.length() > 1) {
             try {
-                text = UTF16.valueOf(Integer.parseInt(trimmed, 16));
+                text = UTF16.valueOf(Integer.parseInt(digits, 16));
             } catch (Exception e) {
             }
         }
-        int cp = UTF16.charAt(text, 0);
+        int[] result = text.codePoints().toArray();
```

Comment:
observation: `.codePoints()` is a Java 8 method on `CharSequence` that returns an `IntStream` (a `Stream` of ints). I didn't know that existed in Java 8.

Reply:
nice